### PR TITLE
JSON-RPC added 'flag' property to GetPropertyValue concerning audio/subtitle

### PR DESCRIPTION
**Pull request by CastagnaIT**
## Description

This adds the possibility of recognizing the type of audio track or subtitle, otherwise you get a multiple list of all the same languages without any possibility of distinction.
Example here: https://paste.kodi.tv/tohicebayi

If possible i would like to do also a backport to leia

## Motivation and Context

The main problem described in detail here: https://forum.kodi.tv/showthread.php?tid=349440
in short: I need to distinguish these tracks in order to be able to reuse and compare them with others, without taking into account the index, becouse change between the videos

## How Has This Been Tested?

Successfully tested on Windows build, this is an example of the "new" output:
`
{
        u 'currentaudiostream': {
            u 'index': 4,
            u 'name': u 'DD+ ',
            u 'language': u 'eng',
            u 'is_impaired': False,
            u 'is_original': True,
            u 'channels': 6,
            u 'codec': u 'eac3',
            u 'is_default': True,
            u 'bitrate': 458752
        },
        u 'subtitleenabled': False,
        u 'currentsubtitle': {
            u 'index': 4,
            u 'name': u '',
            u 'language': u 'eng',
            u 'is_impaired': False,
            u 'is_default': True,
            u 'is_forced': False
        },
`

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [x] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by CastagnaIT** [comment]
all green, WIN UWP error no related

**Comment by rmrector** [comment]
Thanks. Can you split the flags out to individual boolean switches? The number doesn't mean much to JSON-RPC consumers.

Add the new properties to `Player.Audio.Stream` and `Player.Subtitle` in "xbmc/interfaces/json-rpc/schema/types.json".

Update the JSON-RPC version at "xbmc/interfaces/json-rpc/schema/version.txt" - this is a non-breaking addition so update the second number and reset the third, 10.6.1 to 10.7.0 right now.

Backport is unlikely.

**Comment by CastagnaIT** [comment]
I will do the changes, i'm a c# developer, so i will try my best, eventually you will have to be patient to explain to me how to improve

**Comment by CastagnaIT** [comment]
I do the changes

File: xbmc/interfaces/json-rpc/schema/types.json [hunk 1]
```diff
@@ -230,7 +230,10 @@
       "language": { "type": "string", "required": true },
       "codec": { "type": "string", "required": true },
       "bitrate": { "type": "integer", "required": true },
-      "channels": { "type": "integer", "required": true }
+      "channels": { "type": "integer", "required": true },
+      "is_default": { "type": "boolean", "default": false, "required": true },
```
Comment (rmrector) [inline xbmc/interfaces/json-rpc/schema/types.json]: Please remove the "default" field from these new properties, then we'll be good to go.

Comment (CastagnaIT) [reply xbmc/interfaces/json-rpc/schema/types.json]: Done thanks!

**Comment by rmrector** [comment]
jenkins build this please

**Comment by rmrector** [comment]
Sorry, Jenkins had some trouble so merging had to wait a bit, and now there is a merge conflict with the version.txt. Can you rebase this?

**Comment by rmrector** [comment]
The stars aligned and it is merged!

Thanks for your contribution.

**Comment by ksooo** [comment]
Sorry, being late to the party, but as far as I know we do not use '_' as delimer in attribute names. I would appreciate if this could be fixed quickly, for API consistency reasons.

@DaveBlake what's your take on this?

**Comment by ksooo** [comment]
^^^ @DaveBlake

**Comment by ksooo** [comment]
^^^ @DaveBlake

**Comment by ksooo** [comment]
^^^ @DaveTBlake

**Comment by ksooo** [comment]
@rmrector May I kindly ask you not to merge a PR that changes core code without a code review from a Kodi core developer. Not everything the passes runtime test is good code quality wise. Thanks.

**Comment by ksooo** [comment]
@CastangnaIT I would appreciate if you come up with a follow-up PR addressing my comments. I will be happy to assist you in case something is unclear.

**Comment by CastagnaIT** [comment]
@ksooo check the PR

**Comment by DaveTBlake** [comment]
>Sorry, being late to the party, but as far as I know we do not use '_' as delimiter in attribute names. I  would appreciate if this could be fixed quickly, for API consistency reasons.

@ksooo I agree use of '_' in names is not the style for JSON paramaters.

**Comment by ksooo** [comment]
> ksooo check the PR

 which one?

**Comment by rmrector** [comment]
Ya sorry I missed that.

**Pull request merged**